pyproject.toml

```diff
@@ -16,9 +16,10 @@ authors = [
   { name = "The SunPy Community", email = "sunpy@googlegroups.com" },
 ]
 dependencies = [
-  "astropy>=5.0.6,!=5.1.0",
-  "gwcs>=0.18",
-  "numpy>=1.23.0"
+  "astropy>=5.3",
+  "gwcs>=0.20",
+  "numpy>=1.23.0",
```

Comment:
These are some pretty aggressive version bumps and ndcube nominally follows the [SPEC 0](https://scientific-python.org/specs/spec-0000/) policy of supporting dependencies for two years after their release.

If we really can't get the asdf stuff working without them, one option would be to have stricter version requirements in the `asdf` extra so doing `pip install ndcube[asdf]` would require newer versions of things than `pip install ndcube`